src/olympia/amo/sitemap.py

```diff
@@ -469,16 +474,23 @@ def render_index_xml(sitemaps):
 
 
 def get_sitemap_path(section, app, page=1):
-    if section is None or app is None:
-        # If we don't have a section or app, we don't need a complex directory
-        # structure and we can call the first page 'sitemap' for convenience
-        # (it's likely going to be the only page).
-        endpath = str(page) if page != 1 else 'sitemap'
+    if section is None:
+        sitemap_path = 'sitemap.xml'
     else:
-        endpath = id_to_path(page)
-    return os.path.join(
-        settings.SITEMAP_STORAGE_PATH,
-        section or '',
-        app or '',
-        f'{endpath}.xml',
-    )
+        if (section, amo.APPS.get(app, '') if app else app) not in get_sitemaps():
+            raise InvalidSection
+        try:
+            int(page)
```

Comment:
I think you want to assign `page` to the result of that `int()` call to ensure we are only dealing with an integer in the code later. `int()` accepts a bunch of things : floats, integer strings prefixed/suffixed by whitespace or `+`/`-` signs, etc (the following strings do not raise `ValueError`: ` 42`, `42 `, `+42`).

Like now, ultimately we would fail the load the page, but we would still try to pass down an invalid path to nginx.

FWIW my suggestion would be consistent with what [django paginator does](https://github.com/django/django/blob/d6fbfea08d9bae4165df6a0cfbd4520b5e2242b2/django/core/paginator.py#L48).

Reply:
https://github.com/mozilla/addons-server/pull/19888/commits/7d038816b23460486eb110df8131571d09bf1df7